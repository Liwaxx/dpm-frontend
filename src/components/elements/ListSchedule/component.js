import React from 'react';
import Card from '@material-ui/core/Card';
import PropTypes from 'prop-types';
import Text from '../Text';

export default class Component extends React.Component {
  render() {
    const { classes, data={}, } = this.props;
    return (
      // <Card>
      //   {
      //     data ? (
      //       data.map((data,index) => (
      <Card className={classes.container}>
        <Card className={classes.head}>
          <Text className={classes.time} >
            In 45 min
          </Text>
          <Text className={classes.text} >
            {data.dateTime}
          </Text>
        </Card>
        <Card className={classes.content}>
          <Text className={classes.title} >
            {data.description}
          </Text>
          <Text className={classes.location}>
            {data.location}
          </Text>
        </Card>
      </Card>
      //       ))
      //     ) : (<div>Loading</div>)
      //   }
      // </Card>
    );
  }
}

Component.propTypes = {
  classes: PropTypes.object,
  data: PropTypes.object,
};