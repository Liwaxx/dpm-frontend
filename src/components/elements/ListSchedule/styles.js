const styles = {
  container: {
    maxWidth: 350,
    // border: '1px solid #AEB9CB',
    borderRight: '4px solid #54BAED',
    borderRadius: 'none',
    textAlign: 'left',
    boxShadow: 'none',
  },
  head: {
    width: '28%',
    borderRight: '1px solid #AEB9CB',
    boxShadow: 'none',
    float: 'left',
    borderRadius: 'none',
    height: '105px'
  },
  content: {
    width: '72%',
    float: 'left',
    boxShadow: 'none',
    borderRadius: 'none'
  },
  time: {
    backgroundColor: '#F55151',
    color: 'white',
    fontSize: '12px',
    fontWeight: 400,
    padding: '5px 10px'
  },
  title: {
    fontSize: '14px',
    fontWeight: 700,
    lineSpacing: '14px',
    margin: '2% 1% 5% 2%'
  },
  text: {
    fontSize: '14px',
    fontWeight: 400,
    lineHeight: '14px',
    margin: '5% 5% 0% 25%',
  },
  location: {
    fontSize: '14px',
    fontWeight: 400,
    margin: '5% 5% 0% 8%'
  },
};

export default styles;