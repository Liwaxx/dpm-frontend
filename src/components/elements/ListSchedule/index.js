import { withStyles } from '@material-ui/core/styles';
import ListSchedule from './component';
import styles from './styles';

export default withStyles(styles)(ListSchedule);