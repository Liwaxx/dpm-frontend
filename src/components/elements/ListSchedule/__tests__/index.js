import React from 'react';
import renderer from 'react-test-renderer';
import ListSchedule from '../index';

jest.mock('../../Text');

describe('ListSchedule', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ListSchedule />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});