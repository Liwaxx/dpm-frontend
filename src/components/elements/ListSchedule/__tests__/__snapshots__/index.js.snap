// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListSchedule renders correctly 1`] = `
<div
  className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-8 Component-container-1"
>
  <div
    className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-8 Component-head-2"
  />
  <div
    className="MuiPaper-root-9 MuiPaper-elevation1-12 MuiPaper-rounded-10 MuiCard-root-8 Component-content-3"
  />
</div>
`;
